chain/tendermint/src/runtime/generated.rs

```diff
@@ -0,0 +1,693 @@
+}
+
+pub struct AscEventArray(pub(crate) Array<AscPtr<AscEvent>>);
+
+impl AscType for AscEventArray {
+    fn to_asc_bytes(&self) -> Result<Vec<u8>, DeterministicHostError> {
+        self.0.to_asc_bytes()
+    }
+
+    fn from_asc_bytes(
+        asc_obj: &[u8],
+        api_version: &Version,
+    ) -> Result<Self, DeterministicHostError> {
+        Ok(Self(Array::from_asc_bytes(asc_obj, api_version)?))
+    }
+}
+
+impl AscIndexId for AscEventArray {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintArrayEvent;
+}
+
+pub struct AscValidatorUpdateArray(pub(crate) Array<AscPtr<AscValidatorUpdate>>);
+
+impl AscType for AscValidatorUpdateArray {
+    fn to_asc_bytes(&self) -> Result<Vec<u8>, DeterministicHostError> {
+        self.0.to_asc_bytes()
+    }
+
+    fn from_asc_bytes(
+        asc_obj: &[u8],
+        api_version: &Version,
+    ) -> Result<Self, DeterministicHostError> {
+        Ok(Self(Array::from_asc_bytes(asc_obj, api_version)?))
+    }
+}
+
+impl AscIndexId for AscValidatorUpdateArray {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintArrayValidatorUpdate;
+}
+
+pub struct AscBlockIDFlagEnum(pub(crate) AscEnum<AscBlockIDFlag>);
+
+impl AscType for AscBlockIDFlagEnum {
+    fn to_asc_bytes(&self) -> Result<Vec<u8>, DeterministicHostError> {
+        self.0.to_asc_bytes()
+    }
+
+    fn from_asc_bytes(
+        asc_obj: &[u8],
+        api_version: &Version,
+    ) -> Result<Self, DeterministicHostError> {
+        Ok(Self(AscEnum::from_asc_bytes(asc_obj, api_version)?))
+    }
+}
+
+impl AscIndexId for AscBlockIDFlagEnum {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintBlockIDFlagEnum;
+}
+
+pub struct AscSignedMsgTypeEnum(pub(crate) AscEnum<AscSignedMsgType>);
+
+impl AscType for AscSignedMsgTypeEnum {
+    fn to_asc_bytes(&self) -> Result<Vec<u8>, DeterministicHostError> {
+        self.0.to_asc_bytes()
+    }
+
+    fn from_asc_bytes(
+        asc_obj: &[u8],
+        api_version: &Version,
+    ) -> Result<Self, DeterministicHostError> {
+        Ok(Self(AscEnum::from_asc_bytes(asc_obj, api_version)?))
+    }
+}
+
+impl AscIndexId for AscSignedMsgTypeEnum {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::TendermintSignedMsgTypeEnum;
+}
+
+#[repr(u32)]
+#[derive(AscType, Copy, Clone)]
+pub(crate) enum AscSignedMsgType {
+    SignedMsgTypeUnknown,
+    SignedMsgTypePrevote,
+    SignedMsgTypePrecommit,
+    SignedMsgTypeProposal,
+}
+
+impl AscValue for AscSignedMsgType {}
+
+impl Default for AscSignedMsgType {
+    fn default() -> Self {
+        Self::SignedMsgTypeUnknown
+    }
+}
+
+#[repr(u32)]
+#[derive(AscType, Copy, Clone)]
+pub(crate) enum AscBlockIDFlag {
+    BlockIdFlagUnknown,
+    BlockIdFlagAbsent,
+    BlockIdFlagCommit,
+    BlockIdFlagNil,
+}
+
+impl AscValue for AscBlockIDFlag {}
+
+impl Default for AscBlockIDFlag {
```

Comment:
That's an edge case that I need to update the code generator for, enum and their "variant" is not handled correctly, for NEAR I tweaked by hand and did not update the tool.

Because there is a distinction to be made between pure enum and variant enum and the code generator needs to be adapter to account for those.